core/sr-primitives/src/lib.rs

```diff
@@ -526,6 +542,19 @@ impl codec::Encode for ApplyError {
 	}
 }
 
+#[derive(Eq, PartialEq, Clone, Copy, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize))]
+/// Reason why a dispatch call failed
+pub struct DispatchError {
+	/// Module index, matching the metadata module index
+	pub module: i8, // use i8 instead of u8 because u8 is not supported by parity-codec
```

Comment:
Could we just wrap the `u8` in a unit struct wrapper and make a manual encode implementation?